server.go

```diff
@@ -163,21 +163,33 @@ func (s *server) UpdatePin(ctx context.Context, request v0.UpdatePinRequestObjec
 		user = usersResponse.Payload[0]
 	}
 
+	var res v0.UpdatePinResponseObject
+	for _, smartLockDevice := range s.cfg.Nuki.SmartLockDevices {
```

Comment:
I am wondering how we should handle errors here. Right now we would fail on the first error and not continue.

What about the scenario that the first lock already has the pin (for example because you just added a second lock to the config and the user wants to set their pin again, so they have the same pin on both locks). In that case (I think) the first request would already fail. We can never update the 2nd lock.

My suggestion would be using an error group (https://pkg.go.dev/golang.org/x/sync/errgroup) or similar. That would make all requests concurrently and return the first error. I am not sure if there is good reason not to make concurrent requests.

Reply:
ok, I'll look at it

Reply:
@leonnicolas I try, but no idea if it's ok